glslc/src/file_compiler.cc

```diff
@@ -168,8 +197,36 @@ bool FileCompiler::EmitCompiledResult(
     }
   }
 
-  // Write compilation output to output file.
-  out->write(compilation_output.data(), compilation_output.size());
+  // Write compilation output to output file. If an output format for SPIR-V
+  // binary code is specified, it is handled here.
+  switch (binary_emission_format_) {
```

Comment:
This is because we have `Unspecified` case in the switch statement. Probably we can pull that out and use an `if-else` to separate the specified/unspecified cases. Then we can have a assert out of the switch.